### ISSUE TITLE
HELP !! Long endless iterations when invoking the .fit() . Why?...

### ISSUE THREAD

**Issue by gkoumakis** [catboost/catboost#456]
Problem: {problem description}
Long endless iterations when invoking the .fit() on a dataset of the following characteristics:
65499 instances with 31 columns, numeric data
I tried to implement a simple GridSearchCV and it keeps looping again and again without stopping unless I force quit.

Why is that happening? How can I fix it?
( I need this for my masters project)

Many thanks in advance.

Kind regards,
Georgios

catboost version: {version}
catboost: 0.9.1.1

Operating System: {OS}
macOS Sierra, v. 10.12.6

CPU: {CPU}
# GPU: {GPU}

Python: 3.6.3 |Anaconda custom (64-bit)| (default, Oct  6 2017, 12:04:38) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
scipy: 0.19.1
numpy: 1.13.3
matplotlib: 2.1.0
pandas: 0.20.3
sklearn: 0.19.1
statsmodels: 0.8.0

**Comment by gkoumakis** [comment]
Dear Veronika,

First of all I am honored and "shocked" I got the reply from you directly. :))) 
Well done for what you all managed to create in Yandex.
Thank you for your kind reply. As it is the first time I am using CatBoost I am a bit worried I do things right. 

Ultimately, I managed to get a result (and it is a good one I have to admit!). 

However, just to make sure I described the "issue" properly (because some other people might also experience the same thing), I will try to illustrate below what I get in the output of scikit-learn once I invoke CatBoost with GridSearchCV or even the simple cross_val_score(cv=10).

As you can see it starts displaying what is shown below...:

Learning rate set to 0.038859
0:	learn: 0.6640570	total: 102ms	remaining: 1m 42s
1:	learn: 0.6256534	total: 136ms	remaining: 1m 8s
2:	learn: 0.6019732	total: 178ms	remaining: 59.2s
3:	learn: 0.5714615	total: 214ms	remaining: 53.2s
4:	learn: 0.5420312	total: 248ms	remaining: 49.3s
5:	learn: 0.5182350	total: 278ms	remaining: 46.1s
6:	learn: 0.5038807	total: 331ms	remaining: 46.9s
7:	learn: 0.4906923	total: 370ms	remaining: 45.8s
8:	learn: 0.4755457	total: 400ms	remaining: 44.1s
9:	learn: 0.4575779	total: 431ms	remaining: 42.7s
10:	learn: 0.4411784	total: 461ms	remaining: 41.4s
11:	learn: 0.4277362	total: 492ms	remaining: 40.5s
12:	learn: 0.4196359	total: 524ms	remaining: 39.8s

.... and then it goes on reaching up to 999 and then it starts all over again and again and again looping until all the number of folds have been completed (in the current example 10folds). 
Is that normal?...

992:	learn: 0.1529942	total: 44.4s	remaining: 313ms
993:	learn: 0.1529469	total: 44.4s	remaining: 268ms
994:	learn: 0.1529345	total: 44.4s	remaining: 223ms
995:	learn: 0.1528964	total: 44.5s	remaining: 179ms
996:	learn: 0.1528545	total: 44.5s	remaining: 134ms
997:	learn: 0.1528357	total: 44.6s	remaining: 89.3ms
998:	learn: 0.1527767	total: 44.6s	remaining: 44.7ms
999:	learn: 0.1527614	total: 44.7s	remaining: 0us
Learning rate set to 0.038872
0:	learn: 0.6666382	total: 48.3ms	remaining: 48.2s
1:	learn: 0.6417396	total: 91.8ms	remaining: 45.8s
2:	learn: 0.6188735	total: 134ms	remaining: 44.5s
3:	learn: 0.5891225	total: 182ms	remaining: 45.3s
4:	learn: 0.5589437	total: 221ms	remaining: 44s
5:	learn: 0.5423600	total: 263ms	remaining: 43.6s
6:	learn: 0.5162426	total: 303ms	remaining: 43s
7:	learn: 0.4942572	total: 349ms	remaining: 43.3s
8:	learn: 0.4828975	total: 394ms	remaining: 43.4s
.....
9:	learn: 0.4719642	total: 434ms	remaining: 42.9s
996:	learn: 0.1479671	total: 40.3s	remaining: 121ms
997:	learn: 0.1479449	total: 40.3s	remaining: 80.8ms
998:	learn: 0.1478813	total: 40.3s	remaining: 40.4ms
999:	learn: 0.1478535	total: 40.4s	remaining: 0us
0.777702134971

I mean, what I get with the other algorithms is that I just click the "run" button and I just see the value I am interested in at the output section.
For example, when I use GradientBoosting classifier with my training data I see a number on its own like the following:
0.761723091022

I just want to know if this is the normal way of doing things or whether I have messed things up during the CatBoost installation.

Again, many thanks for your kind reply.

Best regards,
George

**Comment by annaveronika** [comment]
Yes, it is normal, you run 1000 iterations of gradient boosting, do this 10 times, then you do the same for next iteration of grid search. I think you can as well use 3 or 5 folds for grid search, it will be much faster, and I doubt that it will give worse results.

As for output, you can set verbose=False in CatBoost parameters (when creating the estimator). Then you will not get output for every boosting iteration.

**Comment by gkoumakis** [comment]
Dear Anna Veronika and Yandex,

I am very much thankful and obliged to you! 
Many thanks for strengthening and restoring my confidence in using the CatBoost. :))) 👍 
You just made me a CatBoost-evangelist.

Keep up the great work you all do there.
Again many thanks for the prompt replies. 
It is highly appreciated.

Best regards,
George

**Comment by annaveronika** [comment]
Very nice to hear that, you are very welcome! I think I can close the issue then.

**Comment by kirtan-desai** [comment]
Hey gkoumakis can you tell me how much total time did it take to train the model? Even I'm currently working on a dataset with 50k rows on a CPU
